synapse/lib/cell.py

```diff
@@ -1560,32 +1560,22 @@ async def _initAhaService(self):
 
         self.ahasvcname = f'{ahaname}.{ahanetw}'
 
-        async def onlink(proxy):
-            while not proxy.isfini:
-                info = await self.getAhaInfo()
+        async def _runAhaRegLoop():
+
+            while not self.isfini:
```

Comment:
Is this your hot loop? It seems like the old code only ran when the onlink event happened (an inbound connection, right?). Where this happens immediately.

Reply:
I don't think so. There is a 10 second timeout in the `proxy()` call for the ahaclient, and then a `self.waitfini(1)` call.

Reply:
Hotloop was a missing return statement, causing unbound object creation + resource exhaustion ( see 594b693 ).